Tasks/AzureRmWebAppDeployment/MSDeployUtility.ts

```diff
@@ -104,21 +104,22 @@ function getMSDeployCmdForLogs(msDeployCmdArgs: string) : string {
     return msDeployCmdArray.join(",");
 }
 
-function runMSDeployCommand(msDeployExePath: string, msDeployCmdArgs: string, azureRMWebAppConnectionDetails) {
+async function runMSDeployCommand(msDeployExePath: string, msDeployCmdArgs: string, azureRMWebAppConnectionDetails) {
     var msDeployCmdForLogs = getMSDeployCmdForLogs(msDeployCmdArgs);
     tl.debug("[command] "+msDeployExePath+" "+msDeployCmdForLogs);
-
+    var defer = Q.defer<String>();
     tl.exec(msDeployExePath, msDeployCmdArgs)
     .then(function(code){
       azureRmUtil.updateDeploymentStatus(azureRMWebAppConnectionDetails, true);
-        tl.debug("MSDeploy executed Successfully !");
-        tl.debug("Return Code : " + code);
+      tl.debug("MSDeploy executed successfully");
+      defer.resolve(code.toString());
     },
     function(error) {
        azureRmUtil.updateDeploymentStatus(azureRMWebAppConnectionDetails, false);
-        tl.error("MSDeploy failed !");
-        onError(error);
+       tl.error("MSDeploy.exe failed with error");
+        defer.reject(error);
```

Comment:
You can also use await here if that util function returns a promise.  If you use 2 awaits, you don't even really need a separate function.  It's mostly promise boiler plate code you're saving.